crates/sui-core/src/accumulators/mod.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use mysten_common::fatal;
+use sui_types::accumulator_event::AccumulatorEvent;
+use sui_types::effects::{
+    AccumulatorAddress, AccumulatorOperation, AccumulatorValue, AccumulatorWriteV1,
+    TransactionEffects, TransactionEffectsAPI,
+};
+use sui_types::programmable_transaction_builder::ProgrammableTransactionBuilder;
+use sui_types::transaction::{Argument, CallArg, ObjectArg, TransactionKind};
+use sui_types::{
+    Identifier, TypeTag, SUI_ACCUMULATOR_ROOT_OBJECT_ID, SUI_FRAMEWORK_ADDRESS,
+    SUI_FRAMEWORK_PACKAGE_ID,
+};
+
+use crate::authority::authority_per_epoch_store::AuthorityPerEpochStore;
+use crate::authority::epoch_start_configuration::EpochStartConfigTrait;
+use crate::execution_cache::TransactionCacheRead;
+
+/// Merged value is the value stored inside accumulator objects.
+/// Each mergable Move type will map to a single variant as its representation.
+///
+/// For instance, Balance<T> stores a single u64 value, so it will map to SumU128.
+/// A clawback Balance<T> will map to SumU128U128 since it also needs to represent
+/// the amount of the balance that has been frozen.
+#[derive(Debug, Copy, Clone)]
+enum MergedValue {
+    SumU128(u128),
+    SumU128U128(u128, u128),
```

Comment:
Should we denote this as supporting clawback? Or give it some name like  `FundsU128`? 🤷